### PR TITLE
Added attribute support

### PR DESCRIPTION
**Pull request by whiskeysierra**
## Description

Adds support for type-safe attributes that can be provided at the call-site of an HTTP request and passed to plugins. Immediate intention is to pass the operation name for open tracing.

## Motivation and Context

Prerequisite for #509 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: riptide-core/src/test/java/org/zalando/riptide/AttributeTest.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+                .andRespond(withSuccess());
+
+        unit.trace("https://api.example.com")
+                .attribute(attribute, "dXNlcjpzZWNyZXQK")
+                .dispatch(series(),
+                        on(SUCCESSFUL).call(pass()))
+                .join();
+
+        server.verify();
+    }
```
Comment (lukasniemeier-zalando) [inline riptide-core/src/test/java/org/zalando/riptide/AttributeTest.java]: may want to add a test verifying behavior if attribute is not given ("`Secret: unknown`")

**Comment by lukasniemeier-zalando** [comment]
:+1:

File: riptide-core/src/test/java/org/zalando/riptide/AttributeTest.java (hunk 1 above)
Comment (whiskeysierra) [reply riptide-core/src/test/java/org/zalando/riptide/AttributeTest.java]: Done

**Comment by lukasniemeier-zalando** [comment]
:+1:

**Pull request merged**